xmtp_mls/src/groups/mls_ext/decrypted_welcome.rs

```diff
@@ -35,16 +37,31 @@ impl DecryptedWelcome {
         provider: &XmtpOpenMlsProvider<C>,
         hpke_public_key: &[u8],
         encrypted_welcome_bytes: &[u8],
+        encrypted_welcome_metadata_bytes: &[u8],
         wrapper_ciphersuite: WrapperAlgorithm,
-    ) -> Result<DecryptedWelcome, GroupError> {
+    ) -> Result<(DecryptedWelcome, Option<WelcomeMetadata>), GroupError> {
         tracing::info!("Trying to decrypt welcome");
         let hash_ref = find_key_package_hash_ref(provider, hpke_public_key)?;
         let private_key = find_private_key(provider, &hash_ref, &wrapper_ciphersuite)?;
-        let welcome_bytes =
-            unwrap_welcome(encrypted_welcome_bytes, &private_key, wrapper_ciphersuite)?;
 
+        let mut welcome_metadata = None;
+
+        let welcome_bytes = unwrap_welcome(
+            encrypted_welcome_bytes,
+            &private_key,
+            wrapper_ciphersuite.clone(),
+        )?;
         let welcome = deserialize_welcome(&welcome_bytes)?;
 
+        if !encrypted_welcome_metadata_bytes.is_empty() {
+            let metadata_bytes = unwrap_welcome(
```

Comment:
@richardhuaaa curious what you think. Given that this metadata is strictly optional, should we fail open here rather than returning on error?